### PR TITLE
Fixed missing attributes

### PR DESCRIPTION
**Pull request by Cybersmurfen**
Attributes Description and Location were missing in the previous version.

### CODE REVIEW

**Comment by Olen** [comment]
Why do you add the `extract_offset` lines?  There should not be a time-offset in the description or location fields?

**Comment by Cybersmurfen** [comment]
> Why do you add the `extract_offset` lines? There should not be a time-offset in the description or location fields

True, I just copied the row above and made the changes. I'm in need of the location and description for my automations. Why break something that isn't broken and so on. I'm not sure if there should be offsets in the summary as well.

**Comment by Olen** [comment]
> > Why do you add the `extract_offset` lines? There should not be a time-offset in the description or location fields
> 
> True, I just copied the row above and made the changes. I'm in need of the location and description for my automations. Why break something that isn't broken and so on. I'm not sure if there should be offsets in the summary as well.

I don't think that will work.
The `extract_offset` function will check if there is an "offset" added to the title of an event.  This is old code. from before you could add triggers based on time-offsets directly in HA.  
It is looking for something like `Event title. Random Stuff. !!-15` and will understand that "!!-15" means trigger "15 minutes before the actual start_time of the event"
It will then replace the title/summary with the text _without_ the offset ("!!-15") to make it prettier.

When you copy that, you will overwrite any offset that is found in the title (_summary_), because there are probably no offset like that in the description - and especially not in the location.

I think the only code you need is the line that adds description and location to the CalendarEvent()

**Comment by Cybersmurfen** [comment]
> > > Why do you add the `extract_offset` lines? There should not be a time-offset in the description or location fields
> > 
> > 
> > True, I just copied the row above and made the changes. I'm in need of the location and description for my automations. Why break something that isn't broken and so on. I'm not sure if there should be offsets in the summary as well.
> 
> I don't think that will work. The `extract_offset` function will check if there is an "offset" added to the title of an event. This is old code. from before you could add triggers based on time-offsets directly in HA. It is looking for something like `Event title. Random Stuff. !!-15` and will understand that "!!-15" means trigger "15 minutes before the actual start_time of the event" It will then replace the title/summary with the text _without_ the offset ("!!-15") to make it prettier.
> 
> When you copy that, you will overwrite any offset that is found in the title (_summary_), because there are probably no offset like that in the description - and especially not in the location.
> 
> I think the only code you need is the line that adds description and location to the CalendarEvent()

Ah, I see. Well, isn't it a better idea to have the offset in the description instead (or both). If you add it to your own calendar you can hide the offset trigger easier and not letting it show up in for example the family calendar. But I agree, offset in the location - well, not really if it is a true location.

**Comment by Olen** [comment]
The "offset" thing was not invented by me or tybritten or the ical-integration.  It was a "hack" in the original google-calendar integration to make a way to trigger automations before the actual start time of an event.  We just adotped it to behave the same way as the google calendar.  
Anyway, I think it is more or less legacy now, because they changed the calendar integration, and you now _can_ trigger automations with an offset directly in the automation config.

If you update the PR and remove the 4 lines before the CalendarEvent()-line, I'll merge it.

File: custom_components/ical/calendar.py [hunk 1]
```diff
@@ -78,8 +78,12 @@ async def async_update(self):
             return
         (summary, offset) = extract_offset(event["summary"], OFFSET)
         event["summary"] = summary
+        (location, offset) = extract_offset(event["location"], OFFSET)
+        event["location"] = location
+        (description, offset) = extract_offset(event["description"], OFFSET)
+        event["description"] = description
```
Comment (Olen) [inline custom_components/ical/calendar.py:84]: These 4 lines should probably be dropped

**Comment by Olen** [comment]
There is also a similar line here:
https://github.com/tybritten/ical-sensor-homeassistant/blob/f5c17bea1fdcad1831317b6b0de63c48a8643736/custom_components/ical/__init__.py#L112

Which should also get description and location I guess

**Comment by Cybersmurfen** [comment]
> There is also a similar line here:
> 
> https://github.com/tybritten/ical-sensor-homeassistant/blob/f5c17bea1fdcad1831317b6b0de63c48a8643736/custom_components/ical/__init__.py#L112
> 
> Which should also get description and location I guess

Without the four lines there won't be any description or location which is the fix. You need to set both location and summary.
The calendar.ical_xxx entity works fine with the four lines added - there is no changes to init as it is not needed. The offset won't be changed for the summary.

**Comment by Olen** [comment]
> > There is also a similar line here:
> > https://github.com/tybritten/ical-sensor-homeassistant/blob/f5c17bea1fdcad1831317b6b0de63c48a8643736/custom_components/ical/__init__.py#L112
> > 
> > Which should also get description and location I guess
> 
> Without the four lines there won't be any description or location which is the fix. You need to set both location and summary. The calendar.ical_xxx entity works fine with the four lines added


I am pretty sure you are wrong.  
The four lines from 81-84 only extracts any "offet" - e.g. "!!<timestamp>" from the end of the strings, and replaces the original value with the same string _without_ the offset.  As there are no such offsets in the description or location fields, the lines are useless, and here they actually cause an error in the logs if I add them.

> - there is no changes to init as it is not needed. The offset won't be changed for the summary.

The `CalendarEvent` line in calendar.py only adds location and description to the _current_ active calendar entry.
The same line in __init__.py also adds location and description to the rest of the events in the calendar (e.g. past and future events) and without that, you can not have a calendar trigger on any other events than the current one.

**Comment by Olen** [comment]
Youc could probably just truncate the description of it is more than X characters (what size "X" is should be determined first).

**Comment by Cybersmurfen** [comment]
@Olen Well, it was a connection issue on the server-side and I'm not sure if it had anything to do with long descriptions anymore. However it would be an idea to check this in the future - if the calendar fails to load, the integration sometimes doesn't create the entities on start and doesn't check again. However, If I reload the integration manually, it fixes this. I'll do a PR on the init file.

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 541, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 739, in async_device_update
    raise exc
  File "/config/custom_components/ical/calendar.py", line 74, in async_update
    await self.ical_events.update()
  File "/config/custom_components/ical/__init__.py", line 130, in update
    async with session.get(self.url) as response:
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 560, in _request
    await resp.start(conn)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client_reqrep.py", line 899, in start
    message, payload = await protocol.read()  # type: ignore[union-attr]
  File "/usr/local/lib/python3.10/site-packages/aiohttp/streams.py", line 616, in read
    await self._waiter
aiohttp.client_exceptions.ClientOSError: [Errno 104] Connection reset by peer

**Pull request closed without merging**